store/database.go

```diff
@@ -0,0 +1,78 @@
+package store
+
+import (
+	"io"
+)
+
+// VersionedReader extends the Reader interface by making all reads versioned.
+type VersionedReader interface {
```

Comment:
I am a little confused, these interfaces are for rootmulti-store?

Reply:
This interface is for the SS implementation. I wonder If I should just move it there?